src/main/java/com/dawn/jat/illuminati/post/service/PostService.java

```diff
@@ -28,44 +27,77 @@
     @Autowired
     private PostSummaryRepository postSummaryRepository;
 
-    public List<PostSummaryEntity> findPostSummary() {
+    @Autowired
+    private ModelMapper modelMapper;
+
+    public List<PostSummaryEntity> getPostSummaries() {
         return postSummaryRepository.findAll();
     }
 
-    public Optional<PostEntity> findBySlug(String slug) {
-        return postRepository.findBySlug(slug);
+    /**
+     * Get Post by Slug.
+     *
+     * @param slug String
+     * @return PostDto
+     */
+    public PostDto getPostBySlug(String slug) {
+        return modelMapper.map(postRepository.findBySlug(slug), PostDto.class);
     }
 
     /**
      * Create new Entity from postDTO receive from client.
+     *
      * @param postDto PostDto
-     * @return PostEntity
+     * @return PostDto
      */
-    public PostEntity create(PostDto postDto) {
+    public PostDto createPost(PostDto postDto) {
         PostEntity post = new PostEntity();
         post = converter.convertPostDtoToEntity(postDto, post);
-        return postRepository.savePost(post);
+        post = postRepository.savePost(post);
+        return modelMapper.map(post, PostDto.class);
     }
 
-    private PostEntity findById(String id) {
-        Optional<PostEntity> postObj = postRepository.findById(id);
-        if (!postObj.isPresent()) {
-            throw new PostNotFoundException("Saved post cannot be found!!!");
+    /**
+     * Save a post info to an existing post.
+     *
+     * @param postDto PostDto
+     * @return PostDto
+     */
+    public PostDto savePost(PostDto postDto) {
+        PostEntity postEntity = null;
+        if(validateExistPost(postDto)) {
+            postEntity = postRepository.findById(postDto.getId()).get();
+            postEntity = converter.convertPostDtoToEntity(postDto, postEntity);
+            postEntity = postRepository.savePost(postEntity);
         }
-        return postObj.get();
+        if (Objects.isNull(postEntity)) {
+            throw new PostCannotBeSavedException("Post can not be saved.");
+        }
+        return postDto;
```

Comment:
same with createPost, we only return the input postDto, which might not contain all information